### PR TITLE
[IA-4767] Remove legacy aou docker image

### PR DESCRIPTION
**Pull request by LizBaldo**
__Jira ticket__: https://broadworkbench.atlassian.net/browse/IA-4767

### What

- Remove the old aou docker image from leonardo now that it is not used anymore by AOU
- Cache version 2.2.7, not 2.2.8 to match what is expected by AOU prod

### What to test

- 

### Who tested and where

- [ ] This change is covered by automated tests 
  - _NB: Rerun automation tests on this PR by commenting `jenkins retest` or `jenkins multi-test`._
- [ ] I validated this change 
- [ ] Primary reviewer validated this change 
- [ ] I validated this change in the dev environment

### CODE REVIEW

File: http/src/main/resources/reference.conf [hunk 1]
```diff
@@ -112,7 +110,7 @@ dataproc {
 }
 
 gce {
-  customGceImage = "projects/broad-dsp-gcr-public/global/images/leo-gce-image-2023-12-01-17-29-26"
```
Comment (Qi77Qi) [inline http/src/main/resources/reference.conf:115]: you even generated new VM images? :)

File: jenkins/dataproc-custom-images/prepare-custom-leonardo-jupyter-dataproc-image.sh [hunk 2]
```diff
@@ -21,14 +21,11 @@ terra_jupyter_r="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-r:2.2.4"
...
 terra_jupyter_hail="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-hail:1.1.8"
 terra_jupyter_gatk="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-gatk:2.3.6"
-terra_jupyter_aou="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-aou:2.2.8"
+terra_jupyter_aou="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-aou:2.2.7"
 welder_server="us.gcr.io/broad-dsp-gcr-public/welder-server:8667bfe"
 openidc_proxy="broadinstitute/openidc-proxy:2.3.1_2"
 anvil_rstudio_bioconductor="us.gcr.io/broad-dsp-gcr-public/anvil-rstudio-bioconductor:3.18.0"
 
-# Note that this is the version used currently by AOU in production!
-terra_jupyter_aou_old="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-aou:2.1.15"
```
Comment (Qi77Qi) [inline jenkins/dataproc-custom-images/prepare-custom-leonardo-jupyter-dataproc-image.sh:30]: I think we still want this one...and update it to `2.2.7`

File: jenkins/gce-custom-images/prepare_gce_image.sh [hunk 3]
```diff
@@ -38,7 +35,7 @@ cryptomining_detector="us.gcr.io/broad-dsp-gcr-public/cryptomining-detector:0.0.
 
 # This array determines which of the above images are baked into the custom image
 # the entry must match the var name above, which must correspond to a valid docker URI
-docker_image_var_names="welder_server terra_jupyter_python terra_jupyter_r terra_jupyter_bioconductor terra_jupyter_gatk terra_jupyter_aou terra_jupyter_aou_old openidc_proxy anvil_rstudio_bioconductor cryptomining_detector cos_gpu_installer google_cloud_toolbox docker_composer docker_composer_with_auth"
```
Comment (Qi77Qi) [inline jenkins/gce-custom-images/prepare_gce_image.sh]: we still need to cache "old". we just need to update "old" to be `2.2.7`

**Review by Qi77Qi — CHANGES_REQUESTED** [review]
great cleanup! one issue I commented on

**Review by Qi77Qi — APPROVED** [review]
:shipit: 🎉

File: http/src/main/resources/reference.conf (hunk 1 above)
Comment (LizBaldo) [reply http/src/main/resources/reference.conf:115]: Yeah I was a bit trigger happy, but the latest ones I generated have both aou images on it 👍

**Comment by LizBaldo** [comment]
jenkins retest

**Comment by LizBaldo** [comment]
jenkins retest

**Comment by LizBaldo** [comment]
jenkins retest

**Comment by LizBaldo** [comment]
I tested on my BEE the couple of cases that failed on jenkins and it works fine there, so if the job keeps failing I might force merge, because this is blocking AOU prod release correct @Qi77Qi?

**Comment by Qi77Qi** [comment]
oh..this is NOT blocking AoU in ANY way...this PR is JUST for cleanup stuff in Leo

But I'm ok with force merging if failures look irrelevant

**Review by lucymcnatt — APPROVED** [review]

**Comment by LizBaldo** [comment]
Oh okay, I thought AOU needed the 2.2.7 image cached, you don't need that for prod to run smoothly without runtime creation timeouts?

**Comment by LizBaldo** [comment]
jenkins retest

**Comment by LizBaldo** [comment]
jenkins retest

**Comment by LizBaldo** [comment]
All tests did pass at one point, it's just flakiness so merging now

**Comment by Qi77Qi** [comment]
2.2.7 is cached today? no?

**Comment by LizBaldo** [comment]
It was not cached no, which is why I force merged this PR

**Comment by LizBaldo** [comment]
2.2.8 was cached, and they are basically the same, so pulling it would be pretty fast anyways, but better cache it

**Comment by Qi77Qi** [comment]
so what's cached today? 2.1.15 and what(I thought it's 2.2.7, which is why we upgraded to 2.2.7 in AoU)?

**Comment by LizBaldo** [comment]
Today it's 2.1.15, and 2.2.8 (the latest), with this PR it's going to be 2.2.7 for prod and 2.2.8 for pre-prod

**Pull request merged**